File: Contracts/AATM.Contracts/Interfaces/Services/ILocalizationService.vb
Public Interface ILocalizationService
    ReadOnly Property IsRightToLeft As Boolean
    Sub AddOrUpdateString(moduleName As String, uiIdentifier As String, originalString As String, languageCode As String, localizedString As String)
    ''' <summary>
    ''' Defines the contract for a translation service.
    ''' </summary>
    Function Translate(ByVal sourceLang As String, ByVal targetLang As String, ByVal textToTranslate As String) As String
    Function GetAvailableLanguages() As List(Of (display As String, code As String))
    Function GetString(uiIdentifier As String, originalString As String) As String
    Function GetLocalizedStrings() As IDictionary(Of String, String)
End Interface

File: Contracts/AATM.Contracts/Interfaces/Services/ILocalizationService.cs
using System.Collections.Generic;

namespace AATM.Contracts.Interfaces.Services
{
    public interface ILocalizationService
    {
        bool IsRightToLeft { get; }
        void AddOrUpdateString(string moduleName, string uiIdentifier, string originalString, string languageCode, string localizedString);
        /// <summary>
    /// Defines the contract for a translation service.
    /// </summary>
        string Translate(string sourceLang, string targetLang, string textToTranslate);
        List<(string display, string code)> GetAvailableLanguages();
        string GetString(string moduleName, string uiIdentifier, string originalString);
        IDictionary<string, string> GetLocalizedStrings();
        void AddString(string moduleName, string text, string languageCode);
        void SetLanguage(string languageCode);
    }
}